python/tests/test_transforms.py

```diff
@@ -318,3 +420,31 @@ def test_void_transform_str():
 
 def test_void_transform_repr():
     assert repr(VoidTransform()) == "VoidTransform()"
+
+
+@pytest.mark.parametrize(
+    "transform,json",
+    [
+        (YearTransform(), '"year"'),
+        (MonthTransform(), '"month"'),
+        (DayTransform(), '"day"'),
+        (HourTransform(), '"hour"'),
+    ],
+)
+def test_datetime_transform_serde(transform, json):
+    assert transform.json() == json
+    assert TestType.parse_raw(json).__root__ == transform
+
+
+@pytest.mark.parametrize(
```

Comment:
A bit of a nit, but I think this test both tests the str and repr, which I think makes sense to split into two tests (one for `repr` and one for `str`.

We haven't discussed this, but I'm not a fan of the whole `parameterize` functionality in pytest mostly because the decorator initializes before the actual tests ran. For example, if there is an error in the `YearTransform()` constructor, it will fail directly. This conflicts a bit with my personal TDD approach to development.

Therefore I just split everything into separate tests: https://github.com/apache/iceberg/blob/master/python/tests/test_types.py#L546
This way you don't have to check on which input the test is failing.

I've also added this to the list of Python topics for the next community sync.

Reply:
+1 for splitting out cases.

Reply:
👌  Split it into two tests. 
Let's chat in the sync meeting to see if we should avoid parameterize at all. I can have followup changes if needed.